File: Data Structures & Algorithms/maximum-subarray/submission-4.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int[] dp = new int[nums.length];
        int res = nums[0];
        for(int i = 0; i < nums.length; i++){
            dp[i] = nums[i];
        }

        for(int j = 1; j < nums.length; j++){
            dp[j] = Math.max(nums[j], dp[j-1]+nums[j]);
            res = Math.max(dp[j], res);
        }

        return res;


    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        alphab = [0]*128
        res = 0
        l = 0
        for r in range(len(s)):
            while alphab[ord('a') - ord(s[r])] != 0 and l<= r:
                alphab[ord('a') - ord(s[l])]-=1
                l+=1
            alphab[ord('a') - ord(s[r])]+=1
            res = max(res, r-l+1)


        return res

                